### ISSUE TITLE
java.lang.ClassNotFoundException: clojure.test

### ISSUE THREAD

**Issue by flosell** [flosell/lambdacd#169]
In some circumstances, `clojure.test` is never loaded, resulting in a compile-error when accessing `lambdacd.presentation.pipeline-structure` which depends on it directly without requiring it: 

```
Compiling lambdacd-pipeline.ui-selection
java.lang.ClassNotFoundException: clojure.test, compiling:(lambdacd/presentation/pipeline_structure.clj:26:5)
Exception in thread "main" java.lang.ClassNotFoundException: clojure.test, compiling:(lambdacd/presentation/pipeline_structure.clj:26:5)
	at clojure.lang.Compiler.analyzeSeq(Compiler.java:6730)
	at clojure.lang.Compiler.analyze(Compiler.java:6524)
	at clojure.lang.Compiler.analyze(Compiler.java:6485)
	at clojure.lang.Compiler$IfExpr$Parser.parse(Compiler.java:2772)
	at clojure.lang.Compiler.analyzeSeq(Compiler.java:6723)
	at clojure.lang.Compiler.analyze(Compiler.java:6524)
	at clojure.lang.Compiler.analyzeSeq(Compiler.java:6711)
	at clojure.lang.Compiler.analyze(Compiler.java:6524)
	at clojure.lang.Compiler.analyze(Compiler.java:6485)
	at clojure.lang.Compiler$IfExpr$Parser.parse(Compiler.java:2785)
	at clojure.lang.Compiler.analyzeSeq(Compiler.java:6723)
	at clojure.lang.Compiler.analyze(Compiler.java:6524)
	at clojure.lang.Compiler.analyzeSeq(Compiler.java:6711)
	at clojure.lang.Compiler.analyze(Compiler.java:6524)
	at clojure.lang.Compiler.analyze(Compiler.java:6485)
	at clojure.lang.Compiler$BodyExpr$Parser.parse(Compiler.java:5861)
	at clojure.lang.Compiler$FnMethod.parse(Compiler.java:5296)
	at clojure.lang.Compiler$FnExpr.parse(Compiler.java:3925)
	at clojure.lang.Compiler.analyzeSeq(Compiler.java:6721)
	at clojure.lang.Compiler.analyze(Compiler.java:6524)
	at clojure.lang.Compiler.analyzeSeq(Compiler.java:6711)
	at clojure.lang.Compiler.analyze(Compiler.java:6524)
	at clojure.lang.Compiler.access$300(Compiler.java:38)
	at clojure.lang.Compiler$DefExpr$Parser.parse(Compiler.java:577)
	at clojure.lang.Compiler.analyzeSeq(Compiler.java:6723)
	at clojure.lang.Compiler.analyze(Compiler.java:6524)
	at clojure.lang.Compiler.analyze(Compiler.java:6485)
	at clojure.lang.Compiler.compile1(Compiler.java:7318)
	at clojure.lang.Compiler.compile(Compiler.java:7390)
	at clojure.lang.RT.compile(RT.java:399)
	at clojure.lang.RT.load(RT.java:444)
	at clojure.lang.RT.load(RT.java:412)
	at clojure.core$load$fn__5448.invoke(core.clj:5866)
	at clojure.core$load.doInvoke(core.clj:5865)
	at clojure.lang.RestFn.invoke(RestFn.java:408)
	at clojure.core$load_one.invoke(core.clj:5671)
	at clojure.core$load_lib$fn__5397.invoke(core.clj:5711)
	at clojure.core$load_lib.doInvoke(core.clj:5710)
	at clojure.lang.RestFn.applyTo(RestFn.java:142)
	at clojure.core$apply.invoke(core.clj:632)
	at clojure.core$load_libs.doInvoke(core.clj:5749)
	at clojure.lang.RestFn.applyTo(RestFn.java:137)
	at clojure.core$apply.invoke(core.clj:632)
	at clojure.core$require.doInvoke(core.clj:5832)
	at clojure.lang.RestFn.invoke(RestFn.java:457)
	at lambdacd.state.core$loading__5340__auto____1450.invoke(core.clj:1)
	at clojure.lang.AFn.applyToHelper(AFn.java:152)
	at clojure.lang.AFn.applyTo(AFn.java:144)
	at clojure.lang.Compiler$InvokeExpr.eval(Compiler.java:3623)
	at clojure.lang.Compiler.compile1(Compiler.java:7323)
	at clojure.lang.Compiler.compile1(Compiler.java:7313)
	at clojure.lang.Compiler.compile(Compiler.java:7390)
	at clojure.lang.RT.compile(RT.java:399)
	at clojure.lang.RT.load(RT.java:444)
	at clojure.lang.RT.load(RT.java:412)
	at clojure.core$load$fn__5448.invoke(core.clj:5866)
	at clojure.core$load.doInvoke(core.clj:5865)
	at clojure.lang.RestFn.invoke(RestFn.java:408)
	at clojure.core$load_one.invoke(core.clj:5671)
	at clojure.core$load_lib$fn__5397.invoke(core.clj:5711)
	at clojure.core$load_lib.doInvoke(core.clj:5710)
	at clojure.lang.RestFn.applyTo(RestFn.java:142)
	at clojure.core$apply.invoke(core.clj:632)
	at clojure.core$load_libs.doInvoke(core.clj:5749)
	at clojure.lang.RestFn.applyTo(RestFn.java:137)
	at clojure.core$apply.invoke(core.clj:632)
	at clojure.core$require.doInvoke(core.clj:5832)
	at clojure.lang.RestFn.invoke(RestFn.java:482)
	at lambdacd.presentation.pipeline_state$loading__5340__auto____663.invoke(pipeline_state.clj:1)
	at clojure.lang.AFn.applyToHelper(AFn.java:152)
	at clojure.lang.AFn.applyTo(AFn.java:144)
	at clojure.lang.Compiler$InvokeExpr.eval(Compiler.java:3623)
	at clojure.lang.Compiler.compile1(Compiler.java:7323)
	at clojure.lang.Compiler.compile1(Compiler.java:7313)
	at clojure.lang.Compiler.compile(Compiler.java:7390)
	at clojure.lang.RT.compile(RT.java:399)
	at clojure.lang.RT.load(RT.java:444)
	at clojure.lang.RT.load(RT.java:412)
	at clojure.core$load$fn__5448.invoke(core.clj:5866)
	at clojure.core$load.doInvoke(core.clj:5865)
	at clojure.lang.RestFn.invoke(RestFn.java:408)
	at clojure.core$load_one.invoke(core.clj:5671)
	at clojure.core$load_lib$fn__5397.invoke(core.clj:5711)
	at clojure.core$load_lib.doInvoke(core.clj:5710)
	at clojure.lang.RestFn.applyTo(RestFn.java:142)
	at clojure.core$apply.invoke(core.clj:632)
	at clojure.core$load_libs.doInvoke(core.clj:5749)
	at clojure.lang.RestFn.applyTo(RestFn.java:137)
	at clojure.core$apply.invoke(core.clj:632)
	at clojure.core$require.doInvoke(core.clj:5832)
	at clojure.lang.RestFn.invoke(RestFn.java:421)
	at lambdaui.common.summaries$loading__5340__auto____661.invoke(summaries.clj:1)
	at clojure.lang.AFn.applyToHelper(AFn.java:152)
	at clojure.lang.AFn.applyTo(AFn.java:144)
	at clojure.lang.Compiler$InvokeExpr.eval(Compiler.java:3623)
	at clojure.lang.Compiler.compile1(Compiler.java:7323)
	at clojure.lang.Compiler.compile1(Compiler.java:7313)
	at clojure.lang.Compiler.compile(Compiler.java:7390)
	at clojure.lang.RT.compile(RT.java:399)
	at clojure.lang.RT.load(RT.java:444)
	at clojure.lang.RT.load(RT.java:412)
	at clojure.core$load$fn__5448.invoke(core.clj:5866)
	at clojure.core$load.doInvoke(core.clj:5865)
	at clojure.lang.RestFn.invoke(RestFn.java:408)
	at clojure.core$load_one.invoke(core.clj:5671)
	at clojure.core$load_lib$fn__5397.invoke(core.clj:5711)
	at clojure.core$load_lib.doInvoke(core.clj:5710)
	at clojure.lang.RestFn.applyTo(RestFn.java:142)
	at clojure.core$apply.invoke(core.clj:632)
	at clojure.core$load_libs.doInvoke(core.clj:5749)
	at clojure.lang.RestFn.applyTo(RestFn.java:137)
	at clojure.core$apply.invoke(core.clj:634)
	at clojure.core$use.doInvoke(core.clj:5843)
	at clojure.lang.RestFn.invoke(RestFn.java:408)
	at lambdaui.api_legacy$loading__5340__auto____659.invoke(api_legacy.clj:1)
	at clojure.lang.AFn.applyToHelper(AFn.java:152)
	at clojure.lang.AFn.applyTo(AFn.java:144)
	at clojure.lang.Compiler$InvokeExpr.eval(Compiler.java:3623)
	at clojure.lang.Compiler.compile1(Compiler.java:7323)
	at clojure.lang.Compiler.compile1(Compiler.java:7313)
	at clojure.lang.Compiler.compile(Compiler.java:7390)
	at clojure.lang.RT.compile(RT.java:399)
	at clojure.lang.RT.load(RT.java:444)
	at clojure.lang.RT.load(RT.java:412)
	at clojure.core$load$fn__5448.invoke(core.clj:5866)
	at clojure.core$load.doInvoke(core.clj:5865)
	at clojure.lang.RestFn.invoke(RestFn.java:408)
	at clojure.core$load_one.invoke(core.clj:5671)
	at clojure.core$load_lib$fn__5397.invoke(core.clj:5711)
	at clojure.core$load_lib.doInvoke(core.clj:5710)
	at clojure.lang.RestFn.applyTo(RestFn.java:142)
	at clojure.core$apply.invoke(core.clj:632)
	at clojure.core$load_libs.doInvoke(core.clj:5749)
	at clojure.lang.RestFn.applyTo(RestFn.java:137)
	at clojure.core$apply.invoke(core.clj:632)
	at clojure.core$require.doInvoke(core.clj:5832)
	at clojure.lang.RestFn.invoke(RestFn.java:805)
	at lambdaui.core$loading__5340__auto____304.invoke(core.clj:1)
	at clojure.lang.AFn.applyToHelper(AFn.java:152)
	at clojure.lang.AFn.applyTo(AFn.java:144)
	at clojure.lang.Compiler$InvokeExpr.eval(Compiler.java:3623)
	at clojure.lang.Compiler.compile1(Compiler.java:7323)
	at clojure.lang.Compiler.compile1(Compiler.java:7313)
	at clojure.lang.Compiler.compile(Compiler.java:7390)
	at clojure.lang.RT.compile(RT.java:399)
	at clojure.lang.RT.load(RT.java:444)
	at clojure.lang.RT.load(RT.java:412)
	at clojure.core$load$fn__5448.invoke(core.clj:5866)
	at clojure.core$load.doInvoke(core.clj:5865)
	at clojure.lang.RestFn.invoke(RestFn.java:408)
	at clojure.core$load_one.invoke(core.clj:5671)
	at clojure.core$load_lib$fn__5397.invoke(core.clj:5711)
	at clojure.core$load_lib.doInvoke(core.clj:5710)
	at clojure.lang.RestFn.applyTo(RestFn.java:142)
	at clojure.core$apply.invoke(core.clj:632)
	at clojure.core$load_libs.doInvoke(core.clj:5749)
	at clojure.lang.RestFn.applyTo(RestFn.java:137)
	at clojure.core$apply.invoke(core.clj:632)
	at clojure.core$require.doInvoke(core.clj:5832)
	at clojure.lang.RestFn.invoke(RestFn.java:457)
	at lambdacd_pipeline.ui_selection$loading__5340__auto____36.invoke(ui_selection.clj:1)
	at clojure.lang.AFn.applyToHelper(AFn.java:152)
	at clojure.lang.AFn.applyTo(AFn.java:144)
	at clojure.lang.Compiler$InvokeExpr.eval(Compiler.java:3623)
	at clojure.lang.Compiler.compile1(Compiler.java:7323)
	at clojure.lang.Compiler.compile1(Compiler.java:7313)
	at clojure.lang.Compiler.compile(Compiler.java:7390)
	at clojure.lang.RT.compile(RT.java:399)
	at clojure.lang.RT.load(RT.java:444)
	at clojure.lang.RT.load(RT.java:412)
	at clojure.core$load$fn__5448.invoke(core.clj:5866)
	at clojure.core$load.doInvoke(core.clj:5865)
	at clojure.lang.RestFn.invoke(RestFn.java:408)
	at clojure.core$load_one.invoke(core.clj:5671)
	at clojure.core$compile$fn__5453.invoke(core.clj:5877)
	at clojure.core$compile.invoke(core.clj:5876)
	at user$eval20$fn__29.invoke(form-init5843256671527169315.clj:1)
	at user$eval20.invoke(form-init5843256671527169315.clj:1)
	at clojure.lang.Compiler.eval(Compiler.java:6782)
	at clojure.lang.Compiler.eval(Compiler.java:6772)
	at clojure.lang.Compiler.eval(Compiler.java:6772)
	at clojure.lang.Compiler.load(Compiler.java:7227)
	at clojure.lang.Compiler.loadFile(Compiler.java:7165)
	at clojure.main$load_script.invoke(main.clj:275)
	at clojure.main$init_opt.invoke(main.clj:280)
	at clojure.main$initialize.invoke(main.clj:308)
	at clojure.main$null_opt.invoke(main.clj:343)
	at clojure.main$main.doInvoke(main.clj:421)
	at clojure.lang.RestFn.invoke(RestFn.java:421)
	at clojure.lang.Var.invoke(Var.java:383)
	at clojure.lang.AFn.applyToHelper(AFn.java:156)
	at clojure.lang.Var.applyTo(Var.java:700)
	at clojure.main.main(main.java:37)
Caused by: java.lang.ClassNotFoundException: clojure.test
	at java.net.URLClassLoader.findClass(URLClassLoader.java:381)
	at clojure.lang.DynamicClassLoader.findClass(DynamicClassLoader.java:69)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
	at clojure.lang.DynamicClassLoader.loadClass(DynamicClassLoader.java:77)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	at java.lang.Class.forName0(Native Method)
	at java.lang.Class.forName(Class.java:348)
	at clojure.lang.RT.classForName(RT.java:2154)
	at clojure.lang.RT.classForName(RT.java:2163)
	at clojure.lang.Compiler$HostExpr.maybeClass(Compiler.java:1017)
	at clojure.lang.Compiler$HostExpr.access$600(Compiler.java:795)
	at clojure.lang.Compiler.macroexpand1(Compiler.java:6662)
	at clojure.lang.Compiler.analyzeSeq(Compiler.java:6709)
	... 192 more
Compilation failed: Subprocess failed
```

**Comment by flosell** [comment]
Workaround: Explicitly require `clojure-test` in the namespace that fails to compile (`lambdacd-pipeline.ui-selection` in the above example): 

```clojure
(:require
    [clojure.test :refer :all]
    [lambdaui.core :as lambdaui]
; ...
```

**Comment by minhtuannguyen** [comment]
hi flosell, could you please make a new release containing  this bugfix?. We are struggling a lot with this bug when building artifacts (lein uberjar). We dont want to add [clojure.test :refer :all] to all of possible namespaces. Thank you so much

**Comment by flosell** [comment]
sorry, totally forgot about releasing this! Released now as [`0.13.2`](https://github.com/flosell/lambdacd/releases/tag/0.13.2)

**Comment by minhtuannguyen** [comment]
thank you :)
